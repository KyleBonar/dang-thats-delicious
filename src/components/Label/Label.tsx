import * as React from "react";

import styled from "../../theme/styled-components";

interface LabelProps {
  children: string | JSX.Element | Array<string | JSX.Element>;
  className?: string;
  htmlFor?: string;
}

const StyledLabel = styled.label`
  text-transform: uppercase;
  color: #676767;
  display: block;
  clear: both;
  font-family: FuturaMedium;
`;

const Label: React.SFC<LabelProps> = props => {
  return (
    <StyledLabel htmlFor={props.htmlFor} className={props.className}>
      {props.children}
    </StyledLabel>
  );
};

export default Label;
