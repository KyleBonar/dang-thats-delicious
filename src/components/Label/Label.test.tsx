import * as React from "react";
import * as enzyme from "enzyme";

import Label from "./Label";
// import { fakeLabels } from "../../utils/testUtils/testUtils";
// import { MockLabel } from "../../utils/testUtils/types";

// const mockLabels = fakeLabels();

describe("<Label />", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(<Label>""</Label>);

    expect(wrapper).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = enzyme.shallow(<Label>""</Label>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correct children", () => {
    const wrapper = enzyme.shallow(<Label>abc</Label>);

    expect(wrapper.props().children).toEqual("abc");
  });
});
