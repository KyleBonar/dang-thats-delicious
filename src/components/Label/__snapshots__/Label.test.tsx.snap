// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> matches snapshot 1`] = `
<styled.label>
  ""
</styled.label>
`;
